cognite/client/_api/datapoints.py

```diff
@@ -96,7 +96,9 @@ def retrieve(
             limit=limit,
         )
         dps_list = fetcher.fetch(query)
-
+        for dp in dps_list:
+            dp._granularity = granularity
```

Comment:
This should be set in the DatapointsFetcher, so that you can use the complete functionality on Datapoints objects returned from .query() aswell. And should have unit tests which make sure they have it set.

Also, can't you do the same thing for ._complete_start, ._complete_end, and ._complete_is_step. Why do those need to be passed to .to_pandas()?

Reply:
- support for query is noted

- complete_is_step we don't want to request if not needed, and then we'd like to request all at once. This is one of the most awkward things of all.
- start/end have to do with Datapoints having different ranges, where count and such should fill it in. what should we set these to in DatapointsFetcher?